### PR TITLE
Interface change for `set_study_directions`

### PR DESCRIPTION
**Pull request by gen740**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

Interface change of optuna https://github.com/optuna/optuna/pull/4189.

## What does this implement/fix? Explain your changes.

Corresponding to https://github.com/optuna/optuna/pull/4189, the `create_new_study` interface well be changed. This commit will apply these changes to this project.

### CODE REVIEW

File: optuna_dashboard/_app.py
Comment (c-bata) [inline optuna_dashboard/_app.py:258]: To support multiple Optuna versions, Optuna Dashboard checks `optuna.__version__` and then calls corresponding storage apis like the following.

https://github.com/optuna/optuna-dashboard/blob/c30fe8234f7ddb2e3206d6fa295a0d174dc9f15d/optuna_dashboard/_app.py#L141-L148

Could you update the implementation to keep backward compatibilities?

**Review by c-bata — CHANGES_REQUESTED** [review]
Thank you for your pull request. I left one change request.

File: optuna_dashboard/_app.py
Comment (gen740) [reply optuna_dashboard/_app.py:258]: OK, I see.

**Review by c-bata — APPROVED** [review]
Thanks for the update. LGTM! I will merge this PR after merged https://github.com/optuna/optuna/pull/4189.

**Pull request merged**
